src/main/groovy/io/seqera/wave/auth/RegistryAuthServiceImpl.groovy

```diff
@@ -66,10 +69,25 @@ class RegistryAuthServiceImpl implements RegistryAuthService {
     private CacheLoader<CacheKey, String> loader = new CacheLoader<CacheKey, String>() {
         @Override
         String load(CacheKey key) throws Exception {
-            return getToken0(key)
+            return getToken(key)
         }
     }
 
+    protected String getToken(CacheKey key){
+        // check if there's a record in the store cache (redis)
+        def result = tokenStore.get(key.toString())
```

Comment:
Not sure about this

Reply:
I have create another method to create test for it

Reply:
you are right 
method executed twice
```
14:17:16.890 [io-executor-thread-2] DEBUG i.s.w.auth.RegistryAuthServiceImpl - Authority token lookup for cachekey: 'RegistryAuthServiceImpl$CacheKey(image:hrma017/dev, auth:RegistryAuth(realm:https://auth.docker.io/token, service:registry.docker.io, type:Bearer, refreshable:true, endpoint:https://auth.docker.io/token?service=registry.docker.io), creds:RegistryCredentials[username=****; password=dck****])' => <TOKEN> [from store]
14:17:17.473 [io-executor-thread-2] DEBUG i.s.w.auth.RegistryAuthServiceImpl - Authority token lookup for cachekey: 'RegistryAuthServiceImpl$CacheKey(image:hrma017/dev, auth:RegistryAuth(realm:https://auth.docker.io/token, service:registry.docker.io, type:Bearer, refreshable:true, endpoint:https://auth.docker.io/token?service=registry.docker.io), creds:RegistryCredentials[username=****; password=****])' => <TOKEN> [from store]
```

Reply:
in first run, its runs twice but in subsequent runs only runs one

Reply:
no even if i dont create getToken method, same thing happens, this is happens only when a new wave instance starts, so i think this is expected

Reply:
Pushing a fix

Reply:
This 49af9372 should solve, but I it will break some tests. please have a look

Reply:
sure, i will fix the tests